src/apiUtils.js

```diff
@@ -0,0 +1,86 @@
+
+let registeredHandlers = [];
+
+/**
+ * Register an event handler for the given event name and for the given list of layers. Valid names are: `loaded`, `updated`.
+ *
+ * The 'loaded' event will be fired when all the layers are loaded (and their 'loaded' events are fired).
+ *
+ * The 'updated' event will be fired whenever one of the layers fired an 'updated' event,
+ * but throttled by requestAnimationFrame to return a maximum of one event per frame.
+ *
+ * @param {string} eventName - Supported event names are 'loaded' and 'updated'
+ * @param {carto.Layer[]} layerList - List of layers
+ *
+ * @memberof carto
+ * @api
+ */
+export function on (eventName, layerList, callback) {
+    let internalCallbacks = [];
+    if (eventName === 'loaded') {
+        const waitingGroup = new Set(layerList);
+        layerList.forEach(layer => {
+            const internalCallback = () => {
+                waitingGroup.delete(layer);
+                if (waitingGroup.size === 0) {
+                    callback();
+                }
+            };
+            layer.on('loaded', internalCallback);
+            internalCallbacks.push(internalCallback);
+        });
+    } else if (eventName === 'updated') {
+        let scheduledRAF = false;
+        layerList.forEach(layer => {
+            const internalCallback = () => {
+                if (!scheduledRAF) {
+                    window.requestAnimationFrame(() => {
+                        scheduledRAF = false;
+                        callback();
+                    });
+                }
+            };
+            layer.on(eventName, internalCallback);
```

Comment:
Instead of `layer.on(eventName...` I would say `layer.on('updated'...` to be consisted with the loaded event.